src/Resolver/Reflector.php

```diff
@@ -129,7 +129,7 @@ public function getTraits($class)
             } while ($class = get_parent_class($class));
 
             // get traits from ancestor traits
-            while (list($trait) = each($traits)) {
+            foreach($traits as &$trait) {
```

Comment:
My first attempt didn't use the `&`. In fact, prefer not to use it.

But running the tests on PHP 7.1 triggered the same errors you're seeing on 5.5 and 5.6.
So while looking at the recursive flow in that code section a colleague suggested the `&` might resolve it. It did ... for PHP 7.1.

Still not understanding why it's breaking on 5.5 and 5.6